pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,716 @@
+		assert.EqualValues(t, expected.Comment, actual.Comment)
+		assert.Equal(t, expected.SecretType, actual.SecretType)
+		assert.EqualValues(t, expected.Username, actual.Username)
+		assert.Equal(t, expected.OauthAccessTokenExpiryTime, actual.OauthAccessTokenExpiryTime)
+		assert.Equal(t, expected.OauthRefreshTokenExpiryTime, actual.OauthRefreshTokenExpiryTime)
+		assert.EqualValues(t, expected.OauthScopes, actual.OauthScopes)
+		assert.EqualValues(t, expected.IntegrationName, actual.IntegrationName)
+	}
+
+	t.Run("Create: secretWithOAuthClientCredentialsFlow", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthClientCredentialsFlowSecretRequest(id, integrationId).
+			WithOauthScopes(sdk.OauthScopesListRequest{OauthScopesList: []sdk.ApiIntegrationScope{{Scope: "foo"}, {Scope: "bar"}}}).
+			WithComment("a").
+			WithIfNotExists(true)
+
+		err := client.Secrets.CreateWithOAuthClientCredentialsFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Secret.DropFunc(t, id))
+
+		assertions.AssertThat(t,
+			objectassert.Secret(t, id).
+				HasName(id.Name()).
+				HasComment("a").
+				HasSecretType("OAUTH2").
+				HasOauthScopes([]string{"foo", "bar"}).
+				HasDatabaseName(id.DatabaseName()).
+				HasSchemaName(id.SchemaName()),
+		)
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:            id.Name(),
+			Comment:         sdk.String("a"),
+			SecretType:      "OAUTH2",
+			OauthScopes:     []string{"foo", "bar"},
+			IntegrationName: sdk.String(integrationId.Name()),
+		})
+	})
+
+	// It is possible to create secret without specifying both refresh token properties and OAuth scopes
+	// Regarding the https://docs.snowflake.com/en/sql-reference/sql/create-secret secret with empty oauth_scopes list should inherit scopes from security_integration, but it does not
+	t.Run("Create: secretWithOAuth - minimal, without token and scopes", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthClientCredentialsFlowSecretRequest(id, integrationId)
+
+		err := client.Secrets.CreateWithOAuthClientCredentialsFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Secret.DropFunc(t, id))
+
+		assertions.AssertThat(t,
+			objectassert.Secret(t, id).
+				HasName(id.Name()).
+				HasDatabaseName(id.DatabaseName()).
+				HasSchemaName(id.SchemaName()),
+		)
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:            id.Name(),
+			SecretType:      "OAUTH2",
+			IntegrationName: sdk.String(integrationId.Name()),
+		})
+	})
+
+	// regarding the https://docs.snowflake.com/en/sql-reference/sql/create-secret secret with empty oauth_scopes list should inherit scopes from security_integration, but it does not
```

Comment:
here and in the previous test - if it does not, then it is a SNowflake problem that should be revisited in the future, right? In such cases we should leave a TODO with issue number assigned.